File: juddi-client/src/main/java/org/apache/juddi/v3/client/transport/wrapper/Inquiry3to2.java
/*
 * Copyright 2014 The Apache Software Foundation.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.juddi.v3.client.transport.wrapper;

import java.rmi.RemoteException;
import java.util.Map;
import javax.xml.ws.Binding;
import javax.xml.ws.BindingProvider;
import javax.xml.ws.EndpointReference;
import javax.xml.ws.soap.SOAPFaultException;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.apache.juddi.v3.client.UDDIServiceV2;
import org.apache.juddi.v3.client.mapping.MapUDDIv2Tov3;
import org.apache.juddi.v3.client.mapping.MapUDDIv3Tov2;
import org.uddi.api_v3.BindingDetail;
import org.uddi.api_v3.BusinessDetail;
import org.uddi.api_v3.BusinessList;
import org.uddi.api_v3.FindBinding;
import org.uddi.api_v3.FindBusiness;
import org.uddi.api_v3.FindRelatedBusinesses;
import org.uddi.api_v3.FindService;
import org.uddi.api_v3.FindTModel;
import org.uddi.api_v3.GetBindingDetail;
import org.uddi.api_v3.GetBusinessDetail;
import org.uddi.api_v3.GetOperationalInfo;
import org.uddi.api_v3.GetServiceDetail;
import org.uddi.api_v3.GetTModelDetail;
import org.uddi.api_v3.OperationalInfo;
import org.uddi.api_v3.OperationalInfos;
import org.uddi.api_v3.RelatedBusinessesList;
import org.uddi.api_v3.ServiceDetail;
import org.uddi.api_v3.ServiceList;
import org.uddi.api_v3.TModelDetail;
import org.uddi.api_v3.TModelList;
import org.uddi.v2_service.DispositionReport;
import org.uddi.v2_service.Inquire;
import org.uddi.v3_service.DispositionReportFaultMessage;
import org.uddi.v3_service.UDDIInquiryPortType;

/**
 * This class provides a wrapper to enable UDDIv3 clients to talk to UDDIv2 servers
 * via JAXWS Transport. It handles all translations for Inquiry service methods.
 * @author <a href="alexoree@apache.org">Alex O'Ree</a>
 * @since 3.2
 */
public class Inquiry3to2 implements UDDIInquiryPortType, BindingProvider {

        private static Log logger = LogFactory.getLog(Inquiry3to2.class);
        Inquire inquiryService = null;

        public Inquiry3to2() {

                UDDIServiceV2 service = new UDDIServiceV2();
                inquiryService = service.getInquire();

        }
        
        public Inquire getUDDIv2WebServiceClient(){
             return inquiryService;
        }

        @Override
        public BindingDetail findBinding(FindBinding body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapBindingDetail(inquiryService.findBinding(MapUDDIv3Tov2.MapFindBinding(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public BusinessList findBusiness(FindBusiness body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        org.uddi.api_v2.FindBusiness MapFindBusiness = MapUDDIv3Tov2.MapFindBusiness(body);
                        org.uddi.api_v2.BusinessList s = inquiryService.findBusiness(MapFindBusiness);
                        return MapUDDIv2Tov3.MapBusinessList(s);
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public RelatedBusinessesList findRelatedBusinesses(FindRelatedBusinesses body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapRelatedBusinessesList(inquiryService.findRelatedBusinesses(MapUDDIv3Tov2.MapFindRelatedBusiness(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public ServiceList findService(FindService body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapServiceList(inquiryService.findService(MapUDDIv3Tov2.MapFindService(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public TModelList findTModel(FindTModel body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapTModelList(inquiryService.findTModel(MapUDDIv3Tov2.MapFindTModel(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public BindingDetail getBindingDetail(GetBindingDetail body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapBindingDetail(inquiryService.getBindingDetail(MapUDDIv3Tov2.MapGetBindingDetail(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public BusinessDetail getBusinessDetail(GetBusinessDetail body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapBusinessDetail(inquiryService.getBusinessDetail(MapUDDIv3Tov2.MapGetBusinessDetail(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        public static final String VERSION = "2.0";

        @Override
        public OperationalInfos getOperationalInfo(GetOperationalInfo body) throws DispositionReportFaultMessage, RemoteException {
                OperationalInfos ret = new OperationalInfos();

                for (int i = 0; i < body.getEntityKey().size(); i++) {
                        OperationalInfo oi = new OperationalInfo();
                        oi.setEntityKey(body.getEntityKey().get(i));
                        try {
                                org.uddi.api_v2.GetBusinessDetail businessDetail = new org.uddi.api_v2.GetBusinessDetail();
                                businessDetail.setGeneric(VERSION);
                                businessDetail.getBusinessKey().add(body.getEntityKey().get(i));
                                org.uddi.api_v2.BusinessDetail z = inquiryService.getBusinessDetail(businessDetail);
                                oi.setNodeID(z.getOperator());
                                oi.setAuthorizedName(z.getBusinessEntity().get(0).getAuthorizedName());
                        } catch (Exception ex) {
                                logger.warn(ex.getMessage(), ex);
                        }
                        if (oi.getAuthorizedName() != null) {
                                continue;
                        }
                        try {
                                org.uddi.api_v2.GetTModelDetail tModelDetail = new org.uddi.api_v2.GetTModelDetail();
                                tModelDetail.setGeneric(VERSION);
                                tModelDetail.getTModelKey().add(body.getEntityKey().get(i));
                                org.uddi.api_v2.TModelDetail z = inquiryService.getTModelDetail(tModelDetail);
                                oi.setNodeID(z.getOperator());
                                oi.setAuthorizedName(z.getTModel().get(0).getAuthorizedName());
                        } catch (Exception ex) {
                                logger.warn(ex.getMessage(), ex);
                        }
                        if (oi.getAuthorizedName() != null) {
                                continue;
                        }
                        try {
                                //get the service
                                org.uddi.api_v2.GetServiceDetail serviceDetail = new org.uddi.api_v2.GetServiceDetail();
                                serviceDetail.setGeneric(VERSION);
                                serviceDetail.getServiceKey().add(body.getEntityKey().get(i));
                                org.uddi.api_v2.ServiceDetail z = inquiryService.getServiceDetail(serviceDetail);
                                oi.setNodeID(z.getOperator());

                                org.uddi.api_v2.GetBusinessDetail businessDetail = new org.uddi.api_v2.GetBusinessDetail();
                                businessDetail.setGeneric(VERSION);
                                //its owning business
                                businessDetail.getBusinessKey().add(z.getBusinessService().get(0).getBusinessKey());
                                org.uddi.api_v2.BusinessDetail z2 = inquiryService.getBusinessDetail(businessDetail);
                                oi.setNodeID(z.getOperator());
                                oi.setAuthorizedName(z2.getBusinessEntity().get(0).getAuthorizedName());
                        } catch (Exception ex) {
                                logger.warn(ex.getMessage(), ex);
                        }
                        ret.getOperationalInfo().add(oi);
                        ret.setTruncated(false);
                }
                return ret;
        }

        @Override
        public ServiceDetail getServiceDetail(GetServiceDetail body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapServiceDetail(inquiryService.getServiceDetail(MapUDDIv3Tov2.MapGetServiceDetail(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public TModelDetail getTModelDetail(GetTModelDetail body) throws DispositionReportFaultMessage, RemoteException {
                try {
                        return MapUDDIv2Tov3.MapTModelDetail(inquiryService.getTModelDetail(MapUDDIv3Tov2.MapGetTModelDetail(body)));
                } catch (DispositionReport ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                } catch (SOAPFaultException ex) {
                        throw MapUDDIv2Tov3.MapException(ex);
                }
        }

        @Override
        public Map<String, Object> getRequestContext() {
                return ((BindingProvider) inquiryService).getRequestContext();
        }

        @Override
        public Map<String, Object> getResponseContext() {
                return ((BindingProvider) inquiryService).getResponseContext();
        }

        @Override
        public Binding getBinding() {
                return ((BindingProvider) inquiryService).getBinding();
        }

        @Override
        public EndpointReference getEndpointReference() {
                return ((BindingProvider) inquiryService).getEndpointReference();
        }

        @Override
        public <T extends EndpointReference> T getEndpointReference(Class<T> clazz) {
                return ((BindingProvider) inquiryService).getEndpointReference(clazz);
        }

}

File: juddi-client.net/juddi-client.net/org.apache.juddi.v3.client.transport.wrapper/Inquiry3to2.cs
using org.apache.juddi.client.org.apache.juddi.v3.client.mapping;
using org.apache.juddi.v3.client.mapping;
using org.uddi.apiv2;
using org.uddi.apiv3;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace org.apache.juddi.v3.client.transport.wrapper
{
    /// <summary>
    /// This class provides a wrapper to enable UDDIv3 clients to talk to UDDIv2 servers
    ///  via JAXWS Transport. It handles all translations for Inquiry service methods.
    ///  @author <a href="alexoree@apache.org">Alex O'Ree</a>
    ///  @since 3.2.1
    /// </summary>
    public class Inquiry3to2 : UDDI_Inquiry_SoapBinding, IDisposable
    {

        private InquireSoap inquiry = new InquireSoap();
        public Inquiry3to2(string endpointURL)
        {
            // TODO: Complete member initialization
            this.Url = endpointURL;
        }

        public Inquiry3to2()
        {

        }

        public  void Dispose()
        {
            if (inquiry != null)
            {
                inquiry.Dispose();
                inquiry = null;
            }
        }

        private void Init()
        {
            inquiry.Url = this.Url;
        }

        public override org.uddi.apiv3.bindingDetail find_binding(org.uddi.apiv3.find_binding find_binding1)
        {
            Init();
            return MapUDDIv2to3.MapBindingDetail(inquiry.find_binding(MapUDDIv3to2.MapFindBinding(find_binding1)));
        }

        public override org.uddi.apiv3.businessList find_business(org.uddi.apiv3.find_business fb)
        {
            Init();
            return MapUDDIv2to3.MapBusinessList(inquiry.find_business(MapUDDIv3to2.MapFindBusiness(fb)));
        }

        public override uddi.apiv3.serviceList find_service(uddi.apiv3.find_service find_service1)
        {
            Init();
            return MapUDDIv2to3.MapServiceList(inquiry.find_service(MapUDDIv3to2.MapFindService(find_service1)));
        }
        public override uddi.apiv3.relatedBusinessesList find_relatedBusinesses(uddi.apiv3.find_relatedBusinesses find_relatedBusinesses1)
        {
            Init();
            return MapUDDIv2to3.MapRelatedBusinessList(inquiry.find_relatedBusinesses(MapUDDIv3to2.MapFindRelatedBusinesses(find_relatedBusinesses1)));
        }
        public override uddi.apiv3.tModelList find_tModel(uddi.apiv3.find_tModel find_tModel1)
        {
            Init();
            return MapUDDIv2to3.MapTModelList(inquiry.find_tModel(MapUDDIv3to2.MapFindTModel(find_tModel1)));
        }
        public override uddi.apiv3.bindingDetail get_bindingDetail(uddi.apiv3.get_bindingDetail get_bindingDetail1)
        {
            Init();
            return MapUDDIv2to3.MapBindingDetail(inquiry.get_bindingDetail(MapUDDIv3to2.MapGetBindingDetail(get_bindingDetail1)));
        }

        public override uddi.apiv3.businessDetail get_businessDetail(uddi.apiv3.get_businessDetail get_businessDetail1)
        {
            Init();
            return MapUDDIv2to3.MapBusinessDetail(inquiry.get_businessDetail(MapUDDIv2to3.MapGetBusinessDetail(get_businessDetail1)));
        }
        public static readonly String VERSION = "2.0";
        public override operationalInfos get_operationalInfo(get_operationalInfo get_operationalInfo1)
        {
            Init();
            operationalInfos ret = new operationalInfos();
            List<operationalInfo> r = new List<operationalInfo>();
            for (int i = 0; i < get_operationalInfo1.entityKey.Length; i++)
            {
                operationalInfo oi = new operationalInfo();
                oi.entityKey=(get_operationalInfo1.entityKey[i]);
                try
                {
                    org.uddi.apiv2.get_businessDetail businessDetail = new org.uddi.apiv2.get_businessDetail();
                    businessDetail.generic=(VERSION);
                    businessDetail.businessKey = new string[]{get_operationalInfo1.entityKey[i]};
                    org.uddi.apiv2.businessDetail z = inquiry.get_businessDetail(businessDetail);
                    oi.nodeID=(z.@operator);
                    oi.authorizedName=(z.businessEntity[0].authorizedName);
                }
                catch (Exception ex)
                {
                }
                if (oi.authorizedName != null)
                {
                    r.Add(oi);
                    continue;
                }
                try
                {
                    org.uddi.apiv2.get_tModelDetail tModelDetail = new org.uddi.apiv2.get_tModelDetail();
                    tModelDetail.generic=(VERSION);
                    tModelDetail.tModelKey = new string[] { get_operationalInfo1.entityKey[i] };
                    org.uddi.apiv2.tModelDetail z = inquiry.get_tModelDetail(tModelDetail);
                    oi.nodeID=(z.@operator);
                    oi.authorizedName=(z.tModel[0].authorizedName);
                }
                catch (Exception ex)
                {
                }
                if (oi.authorizedName != null)
                {
                    r.Add(oi);
                    continue;
                }
                try
                {
                    //get the service
                    org.uddi.apiv2.get_serviceDetail serviceDetail = new org.uddi.apiv2.get_serviceDetail();
                    serviceDetail.generic=(VERSION);
                    serviceDetail.serviceKey = new string[] { get_operationalInfo1.entityKey[i] };
                    org.uddi.apiv2.serviceDetail z = inquiry.get_serviceDetail(serviceDetail);
                    oi.nodeID=(z.@operator);

                    org.uddi.apiv2.get_businessDetail businessDetail = new org.uddi.apiv2.get_businessDetail();
                    businessDetail.generic=(VERSION);
                    //its owning business
                    businessDetail.businessKey = new string[] { z.businessService[0].businessKey };
                    org.uddi.apiv2.businessDetail z2 = inquiry.get_businessDetail(businessDetail);
                    oi.nodeID = (z2.@operator);
                    oi.authorizedName=(z2.businessEntity[0].authorizedName);
                    r.Add(oi);
                }
                catch (Exception ex)
                {
                }

                
            }
            ret.truncated = (false);
            ret.operationalInfo = r.ToArray();
            return ret;


            
        }
        public override uddi.apiv3.serviceDetail get_serviceDetail(uddi.apiv3.get_serviceDetail get_serviceDetail1)
        {
            Init();
            return MapUDDIv2to3.MapServiceDetail(inquiry.get_serviceDetail(MapUDDIv3to2.MapGetServiceDetail(get_serviceDetail1)));
        }
        public override uddi.apiv3.tModelDetail get_tModelDetail(uddi.apiv3.get_tModelDetail get_tModelDetail1)
        {
            Init();
            return MapUDDIv2to3.MapTModelDetail(inquiry.get_tModelDetail(MapUDDIv3to2.MapGetTModelDetail(get_tModelDetail1)));
        }

    }
}
